packages/react/src/ActionList/Item.tsx

```diff
@@ -219,18 +260,46 @@ const UnwrappedItem = <As extends React.ElementType = 'li'>(
     id: itemId,
   }
 
-  const containerProps = _PrivateItemWrapper
-    ? {role: itemRole ? 'none' : undefined, ...props}
-    : // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      (listSemantics && {...menuItemProps, ...props, ref: forwardedRef}) || {}
+  // Props distribution depends on which component pattern is being used:
+  // 1. _PrivateItemWrapper - used by LinkItem
+  // 2. isLinkItem - direct Item usage with link props
+  // 3. default - regular button/div behavior
+
+  // Create a Set for lookup of interactive-only props
+  const interactivePropsSet = new Set<string>(INTERACTIVE_ELEMENT_PROPS)
```

Comment:
This is a longer list than what we checked before, are there any props that were earlier going to the `<li>` that will now end up on `<a>`? Especially in instances where ActionList.Item was used to act like a Link

For example: `ActionList.Item as="a" href=""` or `ActionList.Item onClick={...}`